### PR TITLE
(GSS) [7.59.x] [JBPM-9915] EJBTimerScheduler.localCache contains entries from already fired timers

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9915

cherry-pick: https://github.com/kiegroup/jbpm/commit/0f229c0097ba4c06cfc7b0667fefa48593b011c9

### CODE REVIEW

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java [hunk 1]
```diff
@@ -340,4 +338,9 @@ public TimerJobInstance getTimerByName(String jobName) {
 		return found;
 	}
 
+    public void evictCache(JobHandle jobHandle) {
+        String jobName =  ((EjbGlobalJobHandle) jobHandle).getUuid();
+        logger.debug("Invalidate job {} with job name {} in cache", jobName, localCache.remove(jobName));
```
Comment (gmunozfe) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:343]: Shouldn't remove call be out of the logger invocation? If not in debug level, it won't be executed

**Review by gmunozfe — CHANGES_REQUESTED** [review]

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java (hunk 1 above)
Comment (elguardian) [reply jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:343]: that will be always executed (it is not a function) as you need to pass it as parameter to the debug message. A whole different thing if it is logged or not.

**Review by gmunozfe — APPROVED** [review]
Looks good to me, awesome work @elguardian !

**Review by afalhambra — APPROVED** [review]
LGTM - Same as cherry-pick

**Pull request merged**